code/game/turfs/simulated/floor/fancy_floor.dm

```diff
@@ -113,6 +114,32 @@
 		icon_state = "basalt[rand(0, 12)]"
 		set_basalt_light(src)
 
+
+/turf/open/floor/grass/fakebasalt //Heart is not a real planeteer power
+	name = "aesthetic volcanic flooring"
+	desc = "Safely recreated turf for your hellplanet-scaping"
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "basalt"
+	floor_tile = /obj/item/stack/tile/basalt
+	ore_type = /obj/item/weapon/ore/glass/basalt
+	slowdown = 0
+
+/turf/open/floor/grass/fakebasalt/Initialize()
+	..()
+	if(prob(15))
+		icon_state = "basalt[rand(0, 12)]"
+		set_basalt_light(src)
+
+/turf/open/floor/grass/fakebasalt/attackby(obj/item/C, mob/user, params)
```

Comment:
Jokes aside, to express a point besides grass this is the only other turf to work in this way specifically so its something that can be done later when more tiles of this type are added as to make the whole system consistent.